interface/wx/webview.h

```diff
@@ -460,11 +460,45 @@ class wxWebView : public wxControl
     virtual void Reload(wxWebViewReloadFlags flags = wxWEBVIEW_RELOAD_DEFAULT) = 0;
 
     /**
-        Runs the given javascript code.
-        @note When using wxWEBVIEW_BACKEND_IE you must wait for the current
+        Sets IE emulation level.
+        @param defaultLevel @false to set level to IE7, @true to set level to old mode.
+        @return @true if level can be set, @false if it is not.
+        @note Only avaliable on wxWEBVIEW_BACKEND_IE.
+        @since 3.1.1
+    */
+    virtual bool MSWSetEmulationLevel(bool defaultLevel = false) = 0;
+
+    /**
+        Runs the given JavaScript. It returns true if the script was run successfully and
+        additionally returns the script return value in output.
+        It returns strings, integers, floating point numbers, booleans and objects (as JSON).
+        @note When using wxWEBVIEW_WEBKIT (GTK), it returns true always unless you pass
+              an output param to the method. In that case, it would return false. RunScript
+              outputs are supported on wxWEBVIEW_WEBKIT2 (GTK3).
+              When using wxWEBVIEW_WEBKIT (OSX), there are two limits:
+              1) Javascript allocations greater than 10MB.
+              2) Javascript that takes longer than 10 seconds to execute.
+
+              When using wxWEBVIEW_BACKEND_IE you must wait for the current
               page to finish loading before calling RunScript().
-    */
-    virtual void RunScript(const wxString& javascript) = 0;
+
+              When using wxWEBVIEW_BACKEND_IE, JSON is not available in Quirks or 
+              IE6/7 standards mode, which is unfortunately the default one for the embedded browser control, see
+              https://docs.microsoft.com/en-us/scripting/javascript/reference/json-object-javascript#requirements
+              and see here how to make a program run use "modern" modes
+              https://msdn.microsoft.com/en-us/library/ee330730(v=vs.85)#browser_emulation
+              You can use SetEmulationLevel to change emulation level, for example:
+              @code
+                  SetEmulationLevel();
+              @endcode
+              However, if you don't want to use this, there is an implementation of JSON parser
```

Comment:
This is not really helpful. I'd like to know what exactly needs to be done in order to use a function returning a JS object, could you please provide precise instructions here, i.e. "You have to do this and that under MSW".

Also ideal would be to have the code actually showing how to do this and that in the sample.